test/extended/builds/volumes.go

```diff
@@ -126,3 +133,233 @@ var _ = g.Describe("[sig-builds][Feature:Builds][volumes] build volumes", func()
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				g.By("creating a build config")
+				err = oc1.Run("create").Args("-f", csiDockerBuildConfig).Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				g.By("starting a build and waiting for it to complete")
+				br, _ := exutil.StartBuildAndWait(oc, "mydockertest")
+				br.AssertSuccess()
+
+				g.By("ensuring that the build pod logs contain the provided shared")
+				buildPodLogs, err := br.Logs()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				o.Expect(buildPodLogs).To(o.ContainSubstring("my-secret-value"))
+
+				g.By("creating a deployment config")
+				err = oc1.Run("create").Args("-f", dockerDeploymentConfig).Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				g.By("waiting for the deployment to complete")
+				_, err = deploymentutil.WaitForDeployerToComplete(oc, "mydockertest-1", 5*time.Minute)
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				g.By("ensuring that the shared secret does not exist in the build image")
+				out, err := oc1.Run("rsh").Args("dc/mydockertest", "cat", "/var/run/secrets/some-secret/key").Output()
+				o.Expect(err).To(o.HaveOccurred())
+				o.Expect(out).To(o.ContainSubstring("cat: /var/run/secrets/some-secret/key: No such file or directory"))
+			})
+
+			g.It("should mount given csi shared resource secret without resource refresh into the build pod for docker strategy builds in a techPreview cluster", func() {
+				g.By("creating an imagestream")
+				err := oc1.Run("create").Args("-f", dockerImageStream).Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				g.By("creating a build config")
+				err = oc1.Run("create").Args("-f", csiWithoutResourceRefreshDockerBuildConfig).Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				g.By("starting a build and waiting for it to complete")
+				br, _ := exutil.StartBuildAndWait(oc, "mydockertest")
+				br.AssertSuccess()
+
+				g.By("ensuring that the build pod logs contain the provided shared")
+				buildPodLogs, err := br.Logs()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				o.Expect(buildPodLogs).To(o.ContainSubstring("my-secret-value"))
+
+				g.By("creating a deployment config")
+				err = oc1.Run("create").Args("-f", dockerDeploymentConfig).Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				g.By("waiting for the deployment to complete")
+				_, err = deploymentutil.WaitForDeployerToComplete(oc, "mydockertest-1", 5*time.Minute)
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				g.By("ensuring that the shared secret does not exist in the build image")
+				out, err := oc1.Run("rsh").Args("dc/mydockertest", "cat", "/var/run/secrets/some-secret/key").Output()
+				o.Expect(err).To(o.HaveOccurred())
+				o.Expect(out).To(o.ContainSubstring("cat: /var/run/secrets/some-secret/key: No such file or directory"))
+			})
+		}
+	})
+})
+
+// isNoUpgrade checks if a cluster is a TechPreviewNoUpgrade cluster
+func isNoUpgrade(config *rest.Config) (bool, error) {
```

Comment:
Be explicit that you're checking for `TechPreviewNoUpgrade`. There are other feature sets that end with `NoUpgrade`.

```suggestion
// isTechPreviewNoUpgrade checks if a cluster is a TechPreviewNoUpgrade cluster
func isTechPreviewNoUpgrade(config *rest.Config) (bool, error) {
```

Reply:
fixed